crates/shared/src/gas_price_estimation/mod.rs

```diff
@@ -28,6 +34,7 @@ impl FromStr for GasEstimatorType {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         match s.to_ascii_lowercase().as_str() {
             "web3" => Ok(GasEstimatorType::Web3),
+            "alloy" => Ok(GasEstimatorType::Alloy),
             "native" => Ok(GasEstimatorType::Native),
```

Comment:
I'm curious to know how the native differs from the web3 one, is it just the calculation?

Reply:
`Web3` simply calls `eth_gasPrice` and tries to adapt the legacy gas price into the EIP 1559 format.
`Native` and `Alloy` both use `eth_feeHistory` to check how much recent transactions paid in gas to estimate how big of a tip is likely necessary to ensure fast inclusion. These methods are also "fully aware" of EIP 1559 mechanics.